akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -967,9 +979,10 @@ final class ClusterReceptionist(pubSubMediator: ActorRef, settings: ClusterRecep
       }
 
     case MemberRemoved(m, _) ⇒
-      if (m.address == selfAddress)
+      if (m.address == selfAddress) {
+        clientInteractions.keySet.foreach(_ ! ReceptionistShutdown)
```

Comment:
I'd rather place this in `postStop` in case the receptionist is stopped for some other reason